### PR TITLE
Travis refactor

### LINKED ISSUE #5015: Setup Continuous Integration for MacOS
There is code for MacOS builds, the freshest update being #4996. However, it breaks often, and one can see it by the number of issues in the bug tracker related to MacOS builds. We need to setup TravisCI to test those builds, and preferably build it with and without Vulkan support.

### PR DESCRIPTION
**Pull request by hcorion**
This refactors our Travis build into separate scripts, and moves us to docker, this has a number of benefits:
- Easier organization of build and deployment process.
- Higher stability, by using docker we no longer have the problem of apt server failures, which were the majority of random Travis failures. With this we rely on only two servers, GitHub and Docker Hub.
- The ability to use a newer version of Ubuntu than supported in Travis, or even a different distro if needed. This is a preliminary step towards getting #2746 fixed.
- Extremely minor bump in job completion speed.

We'll be sticking with building the AppImages on 14.04, because it's needed to keep RPCS3 on https://appimage.github.io.

This also enabled the MacOS build and thus Fixes #5015 it also includes Vulkan support via https://github.com/gfx-rs/portability. We've decided against deploying MacOS builds for now, until we fix the major issues.

The scripts to build docker are currently located at https://github.com/hcorion/rpcs3-docker but the repo should probably be moved to the official RPCS3 organization.

A minor change that I made to the build script is moving away from wget entirely, and only using curl.

The MacOS build is currently failing due to the issues mentioned in #5045

### CODE REVIEW

File: .travis/build-mac.bash [hunk 1]
```diff
@@ -0,0 +1,23 @@
+brew install ccache glew ninja qt
+export CCACHE_SLOPPINESS=pch_defines,time_macros
+export CMAKE_PREFIX_PATH=/usr/local/opt/qt5/
+export PATH="/usr/local/opt/ccache/libexec:$PATH"
+
+# Setup vulkan and gfx-rs/portability
+curl -sLO https://github.com/gfx-rs/portability/releases/download/master-latest/gfx-portability-osx-806f0a561f8e1ee79b6d83655137bf59152ec769.zip
```
Comment (leskal) [inline .travis/build-mac.bash]: New link: https://github.com/gfx-rs/portability/releases/download/latest/gfx-portability-macos-latest.zip

File: .travis/build-mac.bash [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+unzip gfx-portability-osx-*.zip
```
Comment (leskal) [inline .travis/build-mac.bash]: gfx-portability-osx-*.zip->gfx-portability-macos-latest.zip

File: .travis/build-mac.bash [hunk 3]
```diff
@@ -0,0 +1,23 @@
+brew install ccache glew ninja qt
+export CCACHE_SLOPPINESS=pch_defines,time_macros
+export CMAKE_PREFIX_PATH=/usr/local/opt/qt5/
+export PATH="/usr/local/opt/ccache/libexec:$PATH"
+
+# Setup vulkan and gfx-rs/portability
+curl -sLO https://github.com/gfx-rs/portability/releases/download/latest/gfx-portability-macos-latest.zip
+unzip gfx-portability-macos-latest.zip
```
Comment (leskal) [inline .travis/build-mac.bash]: unzip -: gfx-portability-macos-latest.zip
Source: https://unix.stackexchange.com/questions/319622/any-way-to-override-path-traversal-protection-in-unzip/319623#319623

File: .travis/build-mac.bash [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+curl -sLO https://github.com/KhronosGroup/Vulkan-Headers/archive/sdk-1.1.82.0.zip
+unzip sdk-*.zip
```
Comment (leskal) [inline .travis/build-mac.bash]: Same here

File: .travis/build-linux.bash [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+shopt -s nocasematch
+
+# Setup Qt variables
+export QT_BASE_DIR=/opt/qt${QTVERMIN}
+export PATH=$QT_BASE_DIR/bin:$PATH
+export LD_LIBRARY_PATH=$QT_BASE_DIR/lib/x86_64-linux-gnu:$QT_BASE_DIR/lib
+
+cd rpcs3
+
+git submodule update --quiet --init asmjit 3rdparty/ffmpeg 3rdparty/pugixml 3rdparty/GSL 3rdparty/libpng 3rdparty/cereal 3rdparty/hidapi 3rdparty/Optional 3rdparty/xxHash 3rdparty/yaml-cpp Vulkan/glslang
```
Comment (leskal) [inline .travis/build-linux.bash]: Remove 3rdparty/Optional as of #5045

File: .travis/build-mac.bash [hunk 6]
```diff
@@ -0,0 +1,23 @@
+brew install ccache glew ninja qt
+export CCACHE_SLOPPINESS=pch_defines,time_macros
+export CMAKE_PREFIX_PATH=/usr/local/opt/qt5/
+export PATH="/usr/local/opt/ccache/libexec:$PATH"
+
+# Setup vulkan and gfx-rs/portability
+curl -sLO https://github.com/gfx-rs/portability/releases/download/latest/gfx-portability-macos-latest.zip
+unzip gfx-portability-macos-latest.zip
+curl -sLO https://github.com/KhronosGroup/Vulkan-Headers/archive/sdk-1.1.82.0.zip
```
Comment (AniLeo) [inline .travis/build-mac.bash]: Should be 1.1.73 for the sake of consistency since we're using those for Windows and Linux currently

**Comment by leskal** [comment]
According to this: https://stackoverflow.com/questions/46859053/stdvisit-for-variant-fails-to-compile-under-clang-5 , the failure of Travis build, is a bug from gcc 7.1: https://bugs.llvm.org/show_bug.cgi?id=33222 So the prob with Travis is not related with this project or at least for what I found...

**Comment by hcorion** [comment]
Yes, it seems clang doesn't like the gcc 7 headers, I've bumped the compiler to gcc 8.

**Comment by oscarbg** [comment]
@hcorion seems Travis already has Xcode 10 support (on 10.13.6) but not 10.14 yet (https://github.com/travis-ci/travis-ci/issues/10155).. do we need travis support for Macos 10.14 or only travis support for Xcode 10? according to https://github.com/RPCS3/rpcs3/pull/5045 seems only Xcode 10 support is needed not 10.14..

**Comment by hcorion** [comment]
We need both XCode 10 and MacOS Mojave, as Apple has locked away newer c++ features in Mojave.

**Pull request merged**